src/lib/snark_work_lib/single_result.ml

```diff
@@ -0,0 +1,41 @@
+open Core_kernel
+
+module Poly = struct
+  [%%versioned
+  module Stable = struct
+    module V1 = struct
+      type ('single_spec, 'proof) t =
+        { spec : 'single_spec
```

Comment:
Maybe we could identify single jobs also by some sort of ID, instead of spec?

Just an idea for future follow-up PR(s)